### PR TITLE
Revamp the Python SQL reading

### PR DESCRIPTION
**Pull request by jeremykubica**
Remove pandas as a dependency by using sqlite3 directly. Also create a method for writing to sqlite. Adds a bunch of tests.

### CODE REVIEW

File: src/pointings_search/pointing_table.py [hunk 1]
```diff
@@ -62,36 +61,51 @@ def from_csv(cls, filename, alt_names=None):
...
-        connection = sqlite3.connect(db_name)
-        pandas_df = pd.read_sql_query(db_query, connection)
-
-        # Convert the data into an astropy Table (from the Panda's table).
-        ap_table = Table.from_pandas(pandas_df)
+        row_cursor = connection.cursor()
+        for row in row_cursor.execute(db_query):
+            for i, key in enumerate(data_dict):
+                data_dict[key].append(row[i])
+        connection.close()
```
Comment (delucchi-cmu) [inline src/pointings_search/pointing_table.py]: I'm happy that the new code closes the connection, but I'd feel better if this did one of those `with` context things, just so the connection close gets called even if there's an error encountered in the intervening code.

File: src/pointings_search/pointing_table.py [hunk 2]
```diff
@@ -332,4 +346,46 @@ def to_csv(self, filename, overwrite=False):
...
+    def to_sqlite(self, db_name, table_name, overwrite=False):
+        """Write the SQL data to a sqlite database potentially
+        overwriting an existing table.
+
+        Parameters
+        ----------
+        db_name : `str`
+            The file location of pointing database.
+        table_name : `str`
+            The table to query from the database.
```
Comment (delucchi-cmu) [inline src/pointings_search/pointing_table.py]:
```suggestion
            The table to write in the database.
```

File: src/pointings_search/pointing_table.py [hunk 3]
```diff
@@ -332,4 +346,46 @@ def to_csv(self, filename, overwrite=False):
...
+        overwrite : `bool`
+            A Boolean indicating whether to overwrite the an existing table.
```
Comment (delucchi-cmu) [inline src/pointings_search/pointing_table.py]:
```suggestion
            A Boolean indicating whether to overwrite any existing table.
```

File: src/pointings_search/pointing_table.py [hunk 4]
```diff
@@ -332,4 +346,46 @@ def to_csv(self, filename, overwrite=False):
...
+
+        # Insert the rows.
+        write_cursor = connection.cursor()
+        for i in range(len(self.pointings)):
+            row_data = ", ".join([str(value) for value in self.pointings[i].values()])
+            write_cursor.execute(f"INSERT INTO {table_name} VALUES ({row_data})")
+            print(f"INSERT INTO {table_name} VALUES ({row_data})")
+
+        # Close the connection.
+        connection.commit()
```
Comment (delucchi-cmu) [inline src/pointings_search/pointing_table.py]: How many rows are you expecting to make at a time? This transaction could get large, and you might need to do it in a few commits?

File: src/pointings_search/pointing_table.py [hunk 5]
```diff
@@ -332,4 +346,46 @@ def to_csv(self, filename, overwrite=False):
...
+            pass
+
+        # Create the table.
+        column_names = ", ".join(self.pointings.columns)
+        connection.execute(f"CREATE TABLE {table_name} ({column_names})")
+        print(f"CREATE TABLE {table_name} ({column_names})")
+
+        # Insert the rows.
+        write_cursor = connection.cursor()
+        for i in range(len(self.pointings)):
```
Comment (delucchi-cmu) [inline src/pointings_search/pointing_table.py]: Does astropy Table not let you do `for pointing in self.pointings:`?

File: tests/pointings_search/test_pointings_table.py [hunk 6]
```diff
@@ -143,6 +143,54 @@ def test_to_csv_cached():
...
+        assert np.allclose(data2.pointings["flux"], data_dict["flux"])
+        assert "unit_vec_x" in data2.pointings.columns
+        assert "unit_vec_y" in data2.pointings.columns
+        assert "unit_vec_z" in data2.pointings.columns
+        assert "earth_vec_x" in data2.pointings.columns
+        assert "earth_vec_y" in data2.pointings.columns
+        assert "earth_vec_z" in data2.pointings.columns
+
+        # Load only a subset of the data.
+        colmaps = {"obstime": "obstime", "ra": "ra", "dec": "dec"}
```
Comment (delucchi-cmu) [inline tests/pointings_search/test_pointings_table.py]: Can you add a test that uses different names for the columns?

File: src/pointings_search/pointing_table.py (hunk 1 above)
Comment (jeremykubica) [reply src/pointings_search/pointing_table.py]: Done (here and below)

File: src/pointings_search/pointing_table.py (hunk 4 above)
Comment (jeremykubica) [reply src/pointings_search/pointing_table.py]: Good call. I added logic to commit every thousand rows.

File: tests/pointings_search/test_pointings_table.py (hunk 6 above)
Comment (jeremykubica) [reply tests/pointings_search/test_pointings_table.py]: Added.

File: src/pointings_search/pointing_table.py (hunk 5 above)
Comment (jeremykubica) [reply src/pointings_search/pointing_table.py]: Fixed.

**Review by delucchi-cmu — APPROVED** [review]

**Pull request merged**